### DISCUSSION TITLE
not sure why my TI training ended without starting to train

### DISCUSSION THREAD

**Discussion by deathtome1998** [bmaltais/kohya_ss#2614] [Q&A]
I'm trying to follow a guide from civitai for making a textual inversion with this tool, and upon hitting the start training button in the UI, it attempted to start.
it loaded the base SDXL model, and the SDXL vae checkpoint. then when it was done with that it gave me a warning that says ignore directory without repeats, which I'm not sure what that means. 
then it lists the folder with my images, says prepare images, then says 0 train images with repeating which I have 25 images in there. next it says 0 reg images then warning no regularization images, which since I'm not sure what they do I didn't give it any nor do I have any to give it. 
then it says [dataset 0] batch size 1 resolution (512, 512) which is correct, but then says enable bucket is false which I did tick the enable buckets box in the UI, so not sure why it says false for that. and it also say network multiplier 1.0 in that same section. 
under that it says [dataset 0] again then loading image sizes. 
the next thing is make buckets, warning min_bucket_reso and max_bucket_reso are ignored if bucket_no_upscale is set, because bucket reso is defined by image size automatically. then number of images (including repeats) /. 
further down it says mean ar error ( without repeats): nan. also below that it says warning clip_skip will be unexpected / SDXL with four question mark boxes in between clip_skip. 
then use template for training captions. is object: true, then it says no data found. please verify arguments. lastly it says training has ended. 
my understanding of the object template for training captions was that I didn't have to make separate caption txt files of my pictures, which I don't know how to do and that it'd just pull the caption from the training pictures filename.
if someone could help me with this I'd be appreciative.

**Comment by deathtome1998** [comment]
still hoping for some help with this. it's been three weeks now and there hasn't been a single comment giving any potential help yet.

**Comment by neepster** [comment]
I think you DO have to make separate text files with the caption data.  Just use notepad and name the file the same name except .txt.  Not only that but you have to hit the Prepare Training Data button and let it do it's own copying/movement of the files to get the training to work.

**Comment by b-fission** [comment]
> I'm trying to follow a guide from civitai for making a textual inversion with this tool, and upon hitting the start training button in the UI, it attempted to start. it loaded the base SDXL model, and the SDXL vae checkpoint. then when it was done with that it gave me a warning that says ignore directory without repeats, which I'm not sure what that means. then it lists the folder with my images, says prepare images, then says 0 train images with repeating which I have 25 images in there.

Your folders need to be named properly. You'll need a main image folder, which I'll name `images`, and inside that will be the concept folder(s).
Those concept folder(s) must be named using the form: `X_concept class` 
`X` is the number of repeats (1 or higher)
`concept` is a prompt word or term you're trying to train
`class` is optional, it's another word that describes what your concept is related to. 

If you do not name the folders properly, they are ignored as the log indicated. Image files and captions go in the subfolders, and not in the main image folder.

As an example...
if I were training a male character named `s00perb` then the class can be `person` or `man`. I might start with 20 repeats; adjust the number accordingly based on your results, or you add/remove images for the concept.

This would be the resulting folder structure.
`C:\where\ever\training\images`
`C:\where\ever\training\images\20_s00perb man`

> next it says 0 reg images then warning no regularization images, which since I'm not sure what they do I didn't give it any nor do I have any to give it.

Not a problem. You can train without regularization images.

> then it says [dataset 0] batch size 1 resolution (512, 512) which is correct,

If you're training on SDXL, the recommended resolution is 1024,1024, not 512,512.

> but then says enable bucket is false which I did tick the enable buckets box in the UI, so not sure why it says false for that. and it also say network multiplier 1.0 in that same section. under that it says [dataset 0] again then loading image sizes. the next thing is make buckets, warning min_bucket_reso and max_bucket_reso are ignored if bucket_no_upscale is set, because bucket reso is defined by image size automatically. then number of images (including repeats) /. further down it says mean ar error ( without repeats): nan. 

Basically, it's glitching out because there are no images to train on. Your folders need to be set up correctly so it will find your images.

> also below that it says warning clip_skip will be unexpected / SDXL with four question mark boxes in between clip_skip.

Probably ignore that for now.

> then use template for training captions. is object: true, then it says no data found. please verify arguments. lastly it says training has ended. my understanding of the object template for training captions was that I didn't have to make separate caption txt files of my pictures, which I don't know how to do and that it'd just pull the caption from the training pictures filename.

Again, it can't find images until your folders are set up correctly.
Yes, captions are optional. You could manage without them if your images are consistent enough to portray the concept/class that you're training.
